File: src/tomek.js/compiler/template/html_node.rb
# To change this template, choose Tools | Templates
# and open the template in the editor.

require 'template/template_node'

class HtmlNode < TemplateNode
  
	@tag
	@namespace
	
	def tag
		@tag
	end
	
	def namespace
		@namespace
	end
	
	def initialize node
    super
		@tag = node.name.to_s
		@namespace = nil
		if ( node.namespace != nil )
			@namespace = node.namespace.href.to_s
		end
  end
	
end

File: src/tomek.js/compiler.js/template/HtmlNode.ts
/// <reference path="../typings/libxmljs/libxmljs.d.ts" />

import libxmljs = require('libxmljs');
import TemplateNode from './TemplateNode';

export default class HtmlNode extends TemplateNode {
		tag: string;
	namespace: string;

	constructor(xmlNode:libxmljs.Element){
		super();
		this.tag = xmlNode.name();
		if ( xmlNode.namespace() !== null ){
			this.namespace = xmlNode.namespace().href();
		}else{
			this.namespace = null;
		}
	}

}